openedx/core/djangoapps/schedules/resolvers.py

```diff
@@ -358,14 +355,14 @@ def schedules_for_bin(self):
         template_context = get_base_template_context(self.site)
         for schedule in schedules:
             enrollment = schedule.enrollment
+            user = enrollment.user
```

Comment:
If you are pulling the user out of the enrollment to improve readability, I would pull the course_id out too. That way, the call site for `get_week_highlights` is contextually separated from enrollments. Otherwise, pass in the enrollment and pull out its pieces in `get_week_highlights`.

Reply:
Maybe we should enforce `COURSE_UPDATE_WAFFLE_FLAG.is_enabled(course_key)` during class construction. Does it make sense to have a `CourseUpdateResolver` if the course updates feature is turned off?